src/mozillavpn.cpp

```diff
@@ -746,13 +746,13 @@ void MozillaVPN::accountChecked(const QByteArray& json) {
     return;
   }
 
+  m_private->m_user.writeSettings();
+  m_private->m_deviceModel.writeSettings();
```

Comment:
A comment here may be useful to explain why the device model is being written before checkCurrentDevice.

Reply:
Good call, thanks. Just added some comments, with a link to the full explanation here.